common/client-core/src/client/base_client/mod.rs

```diff
@@ -614,7 +616,21 @@ where
             store_client_keys(keys, key_store).await?;
         }
 
-        setup_gateway(setup_method, key_store, details_store).await
+        let init_result = setup_gateway(setup_method, key_store, details_store).await?;
+
+        // Should we really set active gateway here? Was it an omission to not set it there before?
+        // Or was it omitted on purpose?
+        set_active_gateway(
```

Comment:
Is this correct? It's not needed for the SDK changes, but it seemed like it should be here? @jstuczyn

Reply:
i think there might be situation where you'd want to register with new gateway but not set it as active immediately, it should be up to the caller to decide that imo